src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,270 @@
+            UInt160 account = new UInt160(args[0].GetSpan());
+            byte[] tokenId = args[1].GetSpan().ToArray();
+            return BalanceOf(engine.Snapshot, account, tokenId);
+        }
+
+        public virtual BigInteger BalanceOf(StoreView snapshot, UInt160 account, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            StorageItem storage = snapshot.Storages.TryGet(CreateOwnershipKey(account, innerKey));
+            if (storage is null) return BigInteger.Zero;
+            return storage.GetInteroperable<UState>().Balance;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.Array, CallFlags.AllowStates, ParameterTypes = new[] { ContractParameterType.Hash160 }, ParameterNames = new[] { "owner" })]
+        public StackItem TokensOf(ApplicationEngine engine, Array args)
+        {
+            UInt160 owner = new UInt160(args[0].GetSpan());
+            return new InteropInterface(TokensOf(engine.Snapshot, owner));
+        }
+
+        public virtual IEnumerator<TState> TokensOf(StoreView snapshot, UInt160 owner)
+        {
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_Ownership, owner).ToArray()).Select(p =>
+            {
+                UInt160 innerKey = new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+                return snapshot.Storages.TryGet(CreateTokenKey(innerKey)).GetInteroperable<TState>();
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.InteropInterface, CallFlags.None, ParameterTypes = new[] { ContractParameterType.ByteArray }, ParameterNames = new[] { "tokenId" })]
+        public StackItem OwnerOf(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenId = args[0].GetSpan().ToArray();
+            return new InteropInterface(OwnerOf(engine.Snapshot, tokenId));
+        }
+
+        public virtual IEnumerator<UInt160> OwnerOf(StoreView snapshot, byte[] tokenId)
+        {
+            UInt160 innerKey = GetInnerKey(tokenId);
+            return snapshot.Storages.Find(CreateStorageKey(Prefix_Ownership, innerKey).ToArray()).Select(p =>
+            {
+                return new UInt160(p.Key.Key.Skip(1 + UInt160.Length).Take(UInt160.Length).ToArray());
+            }).GetEnumerator();
+        }
+
+        [ContractMethod(0_08000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.Hash160, ContractParameterType.Hash160, ContractParameterType.Integer, ContractParameterType.ByteArray }, ParameterNames = new[] { "from", "to", "amount", "tokenId" })]
+        public virtual StackItem Transfer(ApplicationEngine engine, Array args)
+        {
+            UInt160 from = new UInt160(args[0].GetSpan());
+            UInt160 to = new UInt160(args[1].GetSpan());
+            BigInteger amount = args[2].GetBigInteger();
+            byte[] tokenId = args[3].GetSpan().ToArray();
```

Comment:
Check length, maybe we should use UInt160 for tokenId?

Reply:
No. In NNS, name is the tokenid.

Reply:
In this case, why not use `string`?

Reply:
Nep11 has defined  `tokenid`.

```cs
public static bool transfer(byte[] from, byte[] to, BigInteger amount, byte[] tokenid)
```

Reply:
`byte[]` and `string` are the same in NeoVM.

Reply:
Okay, this is not NNS.